File: src/main/java/cn/gov/forestry/common/domain/bo/SystemScheduleJobTypeEnum.java
package cn.gov.forestry.common.domain.bo;

import lombok.Getter;

public enum SystemScheduleJobTypeEnum {
    IMPORT_EXCEL("201","导入excel"),
    IMPORT_SHAPEFILE("202","导入shapefile"),
    IMPORT_GEOTIFF("203","导入geoTiff"),
    TRANSFER_GLYPHS("301","转换glyphs"),
    ;
    @Getter
    private String code;

    @Getter
    private String name;

    SystemScheduleJobTypeEnum(String code, String name){
        this.code = code;
        this.name = name;
    }

    public static Boolean isImportExcel(String code){
        return IMPORT_EXCEL.getCode().equals(code);
    }

    public static Boolean isImportShapefile(String code){
        return IMPORT_SHAPEFILE.getCode().equals(code);
    }

    public static Boolean isImportGeoTiff(String code){
        return IMPORT_GEOTIFF.getCode().equals(code);
    }

    public static SystemScheduleJobTypeEnum getByCode(String code){
        for(SystemScheduleJobTypeEnum jobTypeEnum : SystemScheduleJobTypeEnum.values()){
            if(jobTypeEnum.getCode().equals(code)){
                return jobTypeEnum;
            }
        }
        return null;
    }
}

File: src/main/python/cn/gov/forestry/common/domain/bo/system_schedule_job_type_enum.py
from enum import Enum
from typing import Optional


class SystemScheduleJobTypeEnum(Enum):
    IMPORT_EXCEL = ("201", "导入excel")
    IMPORT_SHAPEFILE = ("202", "导入shapefile")
    IMPORT_GEOTIFF = ("203", "导入geoTiff")
    TRANSFER_GLYPHS = ("301", "转换glyphs")

    def __init__(self, code: str, name: str):
        self.code = code
        self.name = name

    @classmethod
    def isImportExcel(cls, code: Optional[str]) -> bool:
        return code == cls.IMPORT_EXCEL.code

    @classmethod
    def isImportShapefile(cls, code: Optional[str]) -> bool:
        return code == cls.IMPORT_SHAPEFILE.code

    @classmethod
    def isImportGeoTiff(cls, code: Optional[str]) -> bool:
        return code == cls.IMPORT_GEOTIFF.code

    @classmethod
    def getByCode(cls, code: Optional[str]) -> Optional["SystemScheduleJobTypeEnum"]:
        if code is None:
            return None
        for item in cls:
            if item.code == code:
                return item
        return None


__all__ = ["SystemScheduleJobTypeEnum"]
